cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/loadbalancer/AwsLoadBalancer.java

```diff
@@ -0,0 +1,73 @@
+package com.sequenceiq.cloudbreak.cloud.aws.loadbalancer;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class AwsLoadBalancer {
+
+    private static final String LOAD_BALANCER_NAME_PREFIX = "LoadBalancer";
+
+    private final AwsLoadBalancerScheme scheme;
+
+    private final String awsScheme;
+
+    private final List<AwsListener> listeners;
+
+    private final String name;
+
+    private String arn;
+
+    private boolean createListeners;
+
+    public AwsLoadBalancer(AwsLoadBalancerScheme scheme, List<AwsListener> listeners) {
+        this.scheme = scheme;
+        this.awsScheme = scheme.awsScheme();
+        this.listeners = listeners;
+        this.name = getLoadBalancerName(scheme);
+        this.createListeners = false;
+    }
+
+    public AwsLoadBalancerScheme getScheme() {
+        return scheme;
+    }
+
+    public List<AwsListener> getListeners() {
+        return listeners;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getArn() {
+        return arn;
+    }
+
+    public void setArn(String arn) {
+        this.arn = arn;
+    }
+
+    public String getAwsScheme() {
+        return awsScheme;
+    }
+
+    public boolean isCreateListeners() {
```

Comment:
I can't actually change this getter name, because the template files fetch the class members using standardized getter names. If you use anything else, it can't fetch the value properly and the template won't be filled out.

However, I did change it from `createListeners` to `listenerConfigSet`, so the method name sounds nicer.